client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/provision-status/test/index.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { ProvisionStatus } from '..';
+
+describe( 'ProvisionStatus', () => {
+	it( 'should render the first non-successful action', () => {
+		const status = {
+			siteTransfer: 'success',
+			pluginInstallation: 'pending',
+			migrationKey: 'error',
+		};
+
+		render( <ProvisionStatus status={ status } /> );
+
+		expect( screen.getByText( 'Installing the required plugins' ) ).toBeVisible();
+	} );
+
+	it( 'should render the correct action progress number', () => {
+		const status = {
+			siteTransfer: 'success',
+			pluginInstallation: 'pending',
+			migrationKey: 'error',
+		};
+
+		render( <ProvisionStatus status={ status } /> );
+
+		expect( screen.getByText( '2/3' ) ).toBeVisible();
+	} );
+
+	it( 'should render success message when all actions are successul', () => {
+		const status = {
+			siteTransfer: 'success',
+			pluginInstallation: 'success',
+			migrationKey: 'success',
+		};
+
+		render( <ProvisionStatus status={ status } /> );
+
+		expect(
+			screen.getByText(
+				/Your new site is ready! Enter your migration key into your old site to start your migration./
+			)
+		).toBeVisible();
+	} );
+
+	it( 'should render alternative success message when plugins installation is done, but migration key is not available anymore', () => {
+		const status = {
+			siteTransfer: 'success',
+			pluginInstallation: 'success',
+			migrationKey: 'error',
+		};
+
+		render( <ProvisionStatus status={ status } /> );
+
+		expect(
+			screen.getByText(
+				/Your new site is ready! Retrieve your migration key and enter it into your old site to start your migration./
+			)
+		).toBeVisible();
+	} );
+
+	it( 'should render error message when one of ther first two action fails', () => {
+		const status = {
+			siteTransfer: 'success',
+			pluginInstallation: 'error',
+			migrationKey: 'idle',
+		};
+
+		render( <ProvisionStatus status={ status } /> );
+
+		expect(
```

Comment:
Is it a link right? 
If yes, I think can use `screen.getByRole('link', {name:  'Please contact support`})`

Reply:
It doesn't work exactly like that, but works with `name: 'Please contact support (opens in a new tab)'` because the `ExternalLink` link component add `(opens in a new tab)` as children of `<ScreenReaderText>`.

The whole text is actually a `<p>` that has an `<a>`.
What I tried to do is find readable text but, because of markup, jest reads it as distinct elements.
I am actually curious if you have a suggestion to query something like:
`<p>Hello <strong>world</strong>.</p>`. For the user, "Hello world." is a single sentence, but `getByText("Hello world.")` will not capture it.